### PR TITLE
expression: treat AllowedPacketOverflowed as err when enable strict sql mode and in insertStmt|updateStmt|deleteStmt and without ignore

### LINKED ISSUE #29708: TiDB is not resulting error when max_allowed_packet is breached
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
drop table if exists t1;
CREATE TABLE t1 (a text)character set utf8 ;
INSERT INTO t1 VALUES  (REPEAT(0125,200000000));
select * from t1;
```
### 2. What did you expect to see? (Required)
```sql
mysql> INSERT INTO t1 VALUES  (REPEAT(0125,200000000));
ERROR 1301 (HY000): Result of repeat() was larger than max_allowed_packet (2048) - truncated
mysql> select * from t1;
Empty set (0.00 sec)
```
### 3. What did you see instead (Required)
```sql
mysql> INSERT INTO t1 VALUES  (REPEAT(0125,200000000));
Query OK, 1 row affected, 1 warning (0.00 sec)
 Warning | 1301 | Result of repeat() was larger than max_allowed_packet (67108864) - truncated 
mysql> select * from t1;
+------+
| a    |
+------+
| NULL |
+------+
1 row in set (0.00 sec)
```
### 4. What is your TiDB version? (Required)
```sql
 Release Version: v5.4.0-alpha-67-g17edc5758
Edition: Community
Git Commit Hash: 17edc5758fbf865cada7e156318c41d3ee8a7501
Git Branch: master
UTC Build Time: 2021-11-10 20:45:18
GoVersion: go1.17.2
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
```

### PR DESCRIPTION
**Pull request by UBarney**
### What problem does this PR solve?

Issue Number: close #29708

Problem Summary:

### What is changed and how it works?
treat AllowedPacketOverflowed as err when enable strict sql mode and in insertStmt|updateStmt|deleteStmt and without ignore
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by sylzd — COMMENTED** [review]
should vectorized implement changed either?

**Comment by UBarney** [comment]
> should vectorized implement changed either?

Yes. I'll change that

**Comment by UBarney** [comment]
> should vectorized implement changed either?

Don; ptal @sylzd

**Review by sylzd — APPROVED** [review]
LGTM

**Comment by UBarney** [comment]
@guo-shaoge  PTAL

File: expression/integration_test.go [hunk 1]
```diff
@@ -6989,3 +6989,34 @@ func TestIssue30264(t *testing.T) {
...
+	_, err = tk.Exec("insert into t1 select REPEAT(a,200000000) from t1;")
+	require.NotNil(t, err)
+	tk.MustQuery("select a from t1 order by a;").Check([][]interface{}{
+		{"a"},
+		{"b"},
+	})
+
+	// test cast
+	_, err = tk.Exec(`insert into t1 values  (cast("a" as binary(4294967295)));`)
+	require.NotNil(t, err)
```
Comment (guo-shaoge) [inline expression/integration_test.go:7017]: How about add a case for 'insert ignore'.

Comment (guo-shaoge) [reply expression/integration_test.go:7017]: reset LGTM

**Review by guo-shaoge — APPROVED** [review]

**Pull request merged**